### PR TITLE
strip the tag from images that have digest as well

### PR DESCRIPTION
**Pull request by FrimIdan**
e.g. image:tag@sha256:123...

### CODE REVIEW

File: docker/docker.go [hunk 1]
```diff
@@ -138,96 +142,72 @@ func NewImage(conf *Config) (*Image, error) {
...
+		client:   client,
+		os:       "linux",
+		arch:     "amd64",
 	}
 
-	if port != "" {
-		registry = fmt.Sprintf("%s:%s", registry, port)
+	ref, err := reference.ParseNormalizedNamed(conf.ImageName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse image name. name=%v: %v", conf.ImageName, conf.ImageName)
```
Comment (akpsgit) [inline docker/docker.go]: conf.ImageName twice instead of err

File: docker/docker.go [hunk 2]
```diff
@@ -138,96 +142,72 @@ func NewImage(conf *Config) (*Image, error) {
...
+	image.Registry = reference.Domain(ref)
+	// reference.ParseNormalizedNamed setting `defaultDomain`, we need `dockerHub` domain
+	if image.Registry == defaultDomain {
+		image.Registry = dockerHub
 	}
-	if tag == "" {
-		tag = strings.Join(tagParts, ":")
+	image.Name = reference.Path(ref)
+	if canonical, isDigested := ref.(reference.Canonical); isDigested {
+		image.Tag = canonical.Digest().String()
```
Comment (akpsgit) [inline docker/docker.go]: Why a digest is put into a tag?

Comment (FrimIdan) [reply docker/docker.go]: this field is tag or digest - we used it the same to pull the layers from the repo.
URL = \<registry\>\<name\>/manifests/<tag/digest>

Comment (FrimIdan) [reply docker/docker.go]: from https://docs.docker.com/registry/spec/api/
"GET /v2/<name>/manifests/<reference>"
I can rename the field to `reference` if you want

**Review by akpsgit — APPROVED** [review]

**Pull request merged**